sstables/types.hh

```diff
@@ -616,10 +617,15 @@ struct scylla_metadata {
 static constexpr int DEFAULT_CHUNK_SIZE = 65536;
 
 // checksums are generated using adler32 algorithm.
-struct checksum {
+struct checksum : public weakly_referencable<checksum>, enable_lw_shared_from_this<checksum> {
     uint32_t chunk_size;
     utils::chunked_vector<uint32_t> checksums;
 
+    checksum(uint32_t chunk_size = 0, utils::chunked_vector<uint32_t> checksums = {})
+    : chunk_size(chunk_size)
+    , checksums(std::move(checksums))
```

Comment:
Need indent here. But why the constructor anyway? It's not good to have a constructor with so many defaults. Better keep it as a plain struct.

Reply:
Ah, it's for the initializer below. But 0/1/2 parameter constructor (without explicit!) is just lazy.